### PR TITLE
Mcs integration

### PR DESCRIPTION
**Pull request by Praveenrajmani**

### CODE REVIEW

File: examples/minioinstance.yaml [hunk 1]
```diff
@@ -96,16 +107,27 @@ spec:
   #    operator: Equal
   #    value: storage
   ## Add environment variables to be set in MinIO container (https://github.com/minio/minio/tree/master/docs/config)
+  mcs:
+    image: praveenminio/mcs:first
```
Comment (harshavardhana) [inline examples/minioinstance.yaml:111]: use the minio/mcs

File: examples/minioinstance.yaml [hunk 2]
```diff
@@ -96,16 +107,27 @@ spec:
...
+      name: minio-mcs-secret
+    selector:
+      matchLabels:
+        app: mcs # Should match spec.metadata.labels
+    metadata:
+      labels:
+        app: mcs # Should match spec.selector.matchLabels
   env:
     - name: MINIO_BROWSER
       value: "on"
```
Comment (harshavardhana) [inline examples/minioinstance.yaml:123]: This is implicit, we should perhaps mention how to turn if off

File: pkg/constants/constants.go [hunk 3]
```diff
@@ -50,6 +53,9 @@ const DefaultMinIOImage = "minio/minio:RELEASE.2020-04-23T00-58-49Z"
 // MinIOServerName specifies the default container name for MinIOInstance
 const MinIOServerName = "minio"
 
+// MinIOMcsName specifies the default container name for Mcs
+const MinIOMcsName = "minio-mcs"
```
Comment (harshavardhana) [inline pkg/constants/constants.go:57]: this should be just `mcs`?

File: pkg/resources/deployments/deployment.go [hunk 4]
```diff
@@ -26,8 +26,107 @@ import (
...
+			Name: "MCS_SECRET_KEY",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: secretName,
+					},
+					Key: "mcssecretkey",
+				},
+			},
+		}, corev1.EnvVar{
```
Comment (harshavardhana) [inline pkg/resources/deployments/deployment.go:76]: let's sort this properly with MCS_MINIO_SERVER as the first element and then subsequently added SECRET_KEY, ACCESS_KEY and then add the rest

File: pkg/resources/deployments/deployment.go [hunk 5]
```diff
@@ -46,7 +145,14 @@ func NewForCluster(mi *miniov1beta1.MirrorInstance) *appsv1.Deployment {
 		},
 		Spec: appsv1.DeploymentSpec{
 			Replicas: &replicas,
-			Template: corev1.PodTemplateSpec{},
+			Selector: mi.Spec.Mcs.Selector,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: mcsMetadata(mi),
+				Spec: corev1.PodSpec{
+					Containers:    []corev1.Container{mcsContainer(mi, serviceName)},
+					RestartPolicy: "Always",
```
Comment (harshavardhana) [inline pkg/resources/deployments/deployment.go:153]: On-Failure

Comment (harshavardhana) [reply pkg/resources/deployments/deployment.go:153]: mcs is not allowed to exit otherwise, if it does it means there is a bug in mcs lets not mask it.

Comment (Praveenrajmani) [reply pkg/resources/deployments/deployment.go]: On "Always" restart policy is supported by deployments @harshavardhana 
https://github.com/kubernetes/kubernetes/issues/24725

File: examples/minioinstance.yaml [hunk 6]
```diff
@@ -95,6 +107,18 @@ spec:
...
+  #   image: minio/mcs:v0.0.2
+  #   mcsAccessKey: "mcs"
+  #   mcsSecret:
+  #     name: minio-mcs-secret
+  #   selector:
+  #     matchLabels:
+  #       app: mcs # Should match spec.mcs.metadata.labels
+  #   metadata:
+  #     labels:
+  #       app: mcs # Should match spec.mcs.selector.matchLabels
```
Comment (nitisht) [inline examples/minioinstance.yaml:121]: We can un-comment this, i.e. mcs runs by default.

**Pull request merged**